### PR TITLE
SDIT-1085 Various additional NOMIS fields added to migration

### PR DESCRIPTION
**Pull request by andymarke**
The data is as follows:

- Date other parties added to NOMIS
- Comment related to other parties
- Notification list of staff for hearings
- Date evidence is added
- Repair cost
- Guilty evidence and report text for each charge (to be presented as OTHER evidence)
- Hearing representative text

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/adjudications/AdjudicationsMigrationService.kt [hunk 1]
```diff
@@ -231,7 +222,7 @@ private fun Evidence.toEvidence() = MigrateEvidence(
   },
   details = this.detail,
   reporter = this.createdByUsername,
-  dateAdded = this.date.atStartOfDay().format(DateTimeFormatter.ISO_LOCAL_DATE_TIME),
+  dateAdded = this.date.format(DateTimeFormatter.ISO_LOCAL_DATE),
 )
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/adjudications/AdjudicationsMigrationService.kt:226]: I think this must really be a date - swagger is not clear, but naming convention suggest just a date

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/adjudications/AdjudicationsMigrationService.kt [hunk 2]
```diff
@@ -89,77 +95,61 @@ fun AdjudicationChargeResponse.toAdjudication(): AdjudicationMigrateDto =
       offenceCode = this.charge.offence.code,
       offenceDescription = this.charge.offence.description,
     ),
-    witnesses = this.incident.staffWitnesses.map {
-      MigrateWitness(
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/adjudications/AdjudicationsMigrationService.kt:93]: refactored to remove duplicate code below

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/adjudications/AdjudicationsMigrationService.kt [hunk 3]
```diff
@@ -89,77 +95,61 @@ fun AdjudicationChargeResponse.toAdjudication(): AdjudicationMigrateDto =
...
 
+private fun AdjudicationCharge.toEvidence(incident: AdjudicationIncident): List<MigrateEvidence> =
+  this.takeUnless { it.evidence.isNullOrEmpty() && it.reportDetail.isNullOrEmpty() }.let { charge ->
+    return charge?.let {
+      listOf(
+        MigrateEvidence(
+          evidenceCode = OTHER,
+          details = listOfNotNull(
+            charge.evidence.takeUnless { it.isNullOrEmpty() },
+            charge.reportDetail.takeUnless { it.isNullOrEmpty() },
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/adjudications/AdjudicationsMigrationService.kt:120]: also don't use if blank

File: helm_deploy/values-preprod.yaml [hunk 4]
```diff
@@ -17,7 +17,7 @@ generic-service:
     API_BASE_URL_SENTENCING: https://prisoner-nomis-migration-preprod.hmpps.service.justice.gov.uk
     API_BASE_URL_ACTIVITIES: https://activities-api-preprod.prison.service.justice.gov.uk
     API_BASE_URL_NON_ASSOCIATIONS: https://non-associations-api-preprod.hmpps.service.justice.gov.uk
-    API_BASE_URL_ADJUDICATIONS: https://prisoner-nomis-migration-preprod.hmpps.service.justice.gov.uk
+    API_BASE_URL_ADJUDICATIONS: https://manage-adjudications-api-preprod.hmpps.service.justice.gov.uk
     FEATURE_EVENT_SENTENCE_ADJUSTMENT_UPSERTED: false
```
Comment (andymarke) [inline helm_deploy/values-preprod.yaml:21]: set back to real url

**Review by lbennett-moj — APPROVED** [review]

**Pull request merged**